definitions/features/foreman_proxy.rb

```diff
@@ -6,30 +6,34 @@ class Features::ForemanProxy < ForemanMaintain::Feature
     end
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  FOREMAN_PROXY_SETTINGS_PATHS = ['/etc/foreman-proxy/settings.yml',
+                                  '/usr/local/etc/foreman-proxy/settings.yml'].freeze
+  DHCPD_CONFIG_PATHS = ['/etc/dhcp3/dhcpd.conf',
+                        '/etc/dhcp/dhcpd.conf',
+                        '/usr/local/etc/dhcpd.conf'].freeze
+
+  attr_reader :dhcpd_conf_file, :settings_file, :proxy_settings
 
   def initialize
-    @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
-    @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @dhcpd_conf_file = lookup_into(DHCPD_CONFIG_PATHS)
```

Comment:
A more advanced path would be to read `:config` from `dhcp_isc.yml`. The benefit is that you're reading what the proxy would read. `dhcp_isc.yml` can be found in `:settings_directory` which is a setting in `settings.yml`. I do realize you're starting to replicate the proxy configuration class, but it can catch misconfigurations within the proxy.

You may also need to verify when `dhcp_isc.yml` was introduced. IIRC this was 1.11 so older installs will read `:dhcp_config` from `dhcp.yml`. I think this would affect satellite 6.1 but not 6.2.

There's also the `dhcp_remote_isc` plugin which uses `dhcp_remote_isc.yml` instead of `dhcp_isc.yml` but is otherwise similar. You can find out which provider is used in `dhcp.yml` (`:use_provider`).